Assets/Scripts/Reward/RewardManager.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using AchievementM;
+using BTScriptableObject;
+using UI;
+using Common;
+using System;
+using UnityEngine.UI;
+using SaveLoad;
+
+namespace Reward
+{
+    public class RewardManager : Singleton<RewardManager>
```

Comment:
this service violates SRP - its managing the ui and the business logic for rewards

Reply:
Fixed it, created a separate script for reward UI stuff.